src/main/java/io/cdap/plugin/successfactors/source/metadata/SuccessFactorsSchemaGenerator.java

```diff
@@ -0,0 +1,770 @@
+  private List<SuccessFactorsColumnMetadata> buildComplexTypes(String namespace, EdmProperty edmProperty)
+    throws EdmException {
+
+    EdmComplexType complexType = successFactorsServiceHelper.getComplexType(namespace, edmProperty.getName());
+    if (complexType == null) {
+      return Collections.emptyList();
+    }
+
+    List<SuccessFactorsColumnMetadata> columns = new ArrayList<>();
+
+    List<String> propList = complexType.getPropertyNames();
+    for (String prop : propList) {
+      EdmProperty property = ((EdmProperty) complexType.getProperty(prop));
+      columns.add(buildSuccessFactorsColumnMetadata(namespace, property));
+    }
+
+    return columns;
+  }
+
+  /**
+   * Builds schema from the given list of {@code SuccessFactorsColumnMetadata}
+   *
+   * @param columnDetailList {@code SuccessFactorsColumnMetadata}
+   * @return {@code Schema}
+   */
+  private Schema buildSchema(List<SuccessFactorsColumnMetadata> columnDetailList) {
+    List<Schema.Field> outputSchema = columnDetailList.stream()
+      .map(this::buildSchemaField)
+      .collect(Collectors.toList());
+
+    return Schema.recordOf("SuccessFactorsColumnMetadata", outputSchema);
+  }
+
+  /**
+   * Builds Schema field from {@code SuccessFactorsColumnMetadata}
+   *
+   * @param successFactorsColumnDetail {@code SuccessFactorsColumnMetadata}
+   * @return {@code Schema.Field}
+   */
+  private Schema.Field buildSchemaField(SuccessFactorsColumnMetadata successFactorsColumnDetail) {
+    if (successFactorsColumnDetail.containsChild()) {
+      List<Schema.Field> outputSchema = successFactorsColumnDetail.getChildList()
+        .stream()
+        .map(this::buildSchemaField)
+        .collect(Collectors.toList());
+
+      String typeName = buildNestedTypeUniqueName(successFactorsColumnDetail.getName());
+
+      if (successFactorsColumnDetail.getType().equals(EdmTypeKind.ENTITY.name()) &&
+        (successFactorsColumnDetail.getMultiplicityOrdinal() != null &&
+          successFactorsColumnDetail.getMultiplicityOrdinal() > 0)) {
+        // adding 1 to * multiplicity record to ARRAY type
+        return Schema.Field.of(successFactorsColumnDetail.getName(),
+                               Schema.arrayOf(Schema.recordOf(typeName, outputSchema)));
+      }
+
+      // adding 0 to 1 multiplicity record to NULLABLE
+      return Schema.Field.of(successFactorsColumnDetail.getName(),
+                             Schema.nullableOf(Schema.recordOf(typeName, outputSchema)));
+    }
+
+    return Schema.Field.of(successFactorsColumnDetail.getName(), buildRequiredSchemaType(successFactorsColumnDetail));
+  }
+
+  /**
+   * Build and returns the appropriate schema type.
+   *
+   * @param successFactorsColumnDetail {@code SuccessFactorsColumnMetadata}
+   * @return {@code Schema}
+   */
+  private Schema buildRequiredSchemaType(SuccessFactorsColumnMetadata successFactorsColumnDetail) {
+    Schema schemaType = SCHEMA_TYPE_MAPPING.get(successFactorsColumnDetail.getType());
```

Comment:
handle case where mapping doesn't exist.

Reply:
I have added a `null` check for `schemaType` in case no mapping is found. 288b022add0e0a7a19652bd3d4efff4b18f3bf2f

Reply:
Defaulted to String. 52e6b0cd24aa625f1e5f6fdcfa0bf49895640754